pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -465,4 +471,12 @@ public String sendRequestRaw(String url, String query, ObjectNode requestJson, M
       throw new AssertionError("Should not reach this");
     }
   }
+
+  private static String constructQueryExceptionResponse(ProcessingException pe) {
+    try {
+      return new BrokerResponseNative(pe).toJsonString();
+    } catch (IOException ioe) {
+      throw new AssertionError("Should not reach this");
```

Comment:
Don't do assertion error in production code. You may use `Utils.rethrowException(ioe)` to workaround the checked exception if you prefer that way
```suggestion
      throw new RuntimeException("Caught exception serializing the response", ioe);
```

Reply:
yeah Utils rethrow should be the best. although it should never reach here anyway b/c the constructed response will always be a valid json.